### ISSUE TITLE
JSONDecodeError on empty CLOB

### ISSUE THREAD

**Issue by jpic** [oracle/python-oracledb#429]
1. What versions are you using?

19

platform.platform Linux-3.10.0
sys.maxsize > 2**32 True
platform.python_version 3.12.7
oracledb.__version__ 2.5.0

2. Is it an error or a hang or a crash?

Exception crash

4. What error(s) or behavior you are seeing?

When selecting a CLOB column which contains nothing, which I suppose is what the DB considers as an empty string, the driver crashes with JSONDecodeError.

5. Does your application call init_oracle_client()?

No, thin mode

6. Include a runnable Python script that shows the problem.

Working on this, I can't even have github from the machine where I'm having this case, trying to reproduce locally with an express container although I'm not a DBA ...  

Meanwhile, according to the traceback I see, I think this

https://github.com/oracle/python-oracledb/blob/main/src/oracledb/impl/base/cursor.pyx#L134

Should be changed to

```py
try:
    return json.loads(value)
except json.JSONDecodeError:
    return value
```

This will allow fetchall to work even if a CLOB is empty.

**Comment by jpic** [comment]
Confirmed the try/except clause fixes the issue for me: fetchall doesn't crash anymore on empty CLOB

**Comment by anthony-tuininga** [comment]
Thanks for the report. That probably isn't the right solution. Perhaps this?

```python
if value:
    return json.loads(value)
```

**Comment by anthony-tuininga** [comment]
I have pushed a patch that corrects this issue and have initated a [build](https://github.com/oracle/python-oracledb/actions/runs/12204534610) from which you can download pre-built development wheels once it completes. You can also [build from source](https://python-oracledb.readthedocs.io/en/latest/user_guide/installation.html#installing-from-source-code) if you prefer. If you can test your scenario and confirm the patch works as expected, that would be appreciated!

**Comment by jpic** [comment]
Lovely, thank you very much!

bdist_wheel worked like a charm with GCC 4.9

Your solution is probably better, I don't know CLOB well enough, is there
any db level constraint preventing it from storing any kind of value which
would cause a Json decode error? Otherwise, if value will do just fine

Le ven. 6 déc. 2024, 20:02, Anthony Tuininga ***@***.***> a
écrit :

**Comment by anthony-tuininga** [comment]
The `IS JSON` constraint prevents any value that is not JSON as far as I know -- except for this edge case, of course! :-)

**Comment by jpic** [comment]
Perfect! Is this going to be released in 2.5.1?

**Comment by anthony-tuininga** [comment]
That is indeed the plan!

**Comment by jpic** [comment]
Awesome keep up teh gr8 work m8 🎩

Le sam. 7 déc. 2024, 16:32, Anthony Tuininga ***@***.***> a
écrit :

**Comment by anthony-tuininga** [comment]
The patch has been included in version 2.5.1 which was just released.
